src/lib.rs

```diff
@@ -1,60 +1,64 @@
-extern crate pairing;
-extern crate rand;
-
-use pairing::{CurveAffine, CurveProjective, Engine, Field};
-use rand::{Rand, Rng};
+use pairing::{
+    group::{ff::Field, prime::PrimeCurveAffine, Group},
+    Engine,
+};
+use rand_core::RngCore;
 use std::collections::HashSet;
 
-const HASH_KEY: &[u8] = b"BLSSignatureSeed";
+pub trait BlsEngine: Engine {
+    fn hash_message(message: &[u8]) -> Self::G1Affine;
+}
```

Comment:
This is temporary until the `HashToCurve` traits in `bls12_381` or similar are extracted out into the `group` crate.